### PR TITLE
[ODIN-301] shipping prebuilt binaries 

### PR DESCRIPTION
**Pull request by rostislavdeepcrawl**
This PR contains:
- `publish` now builds binaries (the addon + duckdb with parquet extension) against the following targets: NAPI v4 Darwin (when building on mac) and NAPIv4 Linux (using docker image) and publishes to github releases
- `install` now installs those binaries from releases when appropriate binaries are there (must match node-duckdb version, platform and architecture -- not sure if it checks what NAPI versions are supported by client) or builds them from source if they are not available

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,14 +3,19 @@ cmake_policy(SET CMP0042 NEW)
...
+file(COPY ${DUCKDBIN}  DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/Release)
 include_directories(${CMAKE_JS_INC} ./duckdb/src/include ./duckdb/extension/parquet/include)
-link_directories(./duckdb/build/release/src)
-link_directories(./duckdb/build/release/extension/parquet)
+link_directories(${CMAKE_CURRENT_BINARY_DIR}/Release)
 file(GLOB SOURCE_FILES "./addon/*")
 add_library(${PROJECT_NAME} SHARED ${SOURCE_FILES} ${CMAKE_JS_SRC})
 set_target_properties(${PROJECT_NAME} PROPERTIES PREFIX "" SUFFIX ".node")
 target_link_libraries(${PROJECT_NAME} ${CMAKE_JS_LIB} duckdb parquet_extension)
+set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS "-Wl,-rpath,'$ORIGIN'")
```
Comment (rostislavdeepcrawl) [inline CMakeLists.txt:18]: This sets rpath binary header variable to point to $ORIGIN which is a special value meaning the directory of the binary

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+version: "3"
+services:
+  linux-build:
+    container_name: linux-build
+    working_dir: /root/node-duckdb
+    tty: true
+    stdin_open: true
+    image: 059037012730.dkr.ecr.us-east-1.amazonaws.com/deepcrawl/odin-api:latest
```
Comment (dforsber) [inline docker-compose.yml:8]: is this public image?

Comment (rostislavdeepcrawl) [reply docker-compose.yml:8]: It's in our ECR (dev)

Comment (dforsber) [reply docker-compose.yml:8]: think that is ok for now at least 👍🏻

**Review by dforsber — APPROVED** [review]

**Pull request merged**